pandas/tests/reshape/test_concat.py

```diff
@@ -1279,6 +1279,48 @@ def test_concat_ignore_index(self, sort):
 
         tm.assert_frame_equal(v1, expected)
 
+    def test_concat_same_index_names(self):
+        # GH13475
+        index = pd.Index(list("abcd"))
+
+        frame1 = pd.DataFrame({"x": range(4)}, index=index)
```

Comment:
can you parameterize this, its very hard to understand the cases otherwise.

Reply:
sorry, i'm new to testing. i made the parameters a list of input index names and expected resulting name. i considered using product to test every three index combination, but at that point i'd basically have to replicate much of the logic in the code itself.